quaistats/quaistats.go

```diff
@@ -844,25 +858,62 @@ func (s *Service) reportStats(conn *connWrapper) error {
 		sync := s.backend.Downloader().Progress()
 		syncing = header.Number().Uint64() >= sync.HighestBlock
 	}
+
+	cpuPercent, err := cpu.Percent(0, false)
+	if err != nil {
+		log.Warn("Error getting CPU usage:", err)
+		cpuPercent = []float64{-1.0}
+	}
+
+	// Get RAM usage
+	vmStat, err := mem.VirtualMemory()
+	if err != nil {
+		log.Warn("Error getting RAM usage:", err)
+		vmStat = &mem.VirtualMemoryStat{UsedPercent: -1.0}
+	}
+
+	// Get swap usage
+	swapStat, err := mem.SwapMemory()
+	if err != nil {
+		log.Warn("Error getting swap usage:", err)
+		swapStat = &mem.SwapMemoryStat{UsedPercent: -1.0}
+	}
+
+	// Get disk usage
+	diskUsage, err := dirSize(s.instanceDir)
+	if err != nil {
+		log.Warn("Error calculating directory sizes:", err)
+		diskUsage = -1
+	}
+
 	// Assemble the node stats and send it to the server
 	log.Trace("Sending node details to quaistats")
 
 	stats := map[string]interface{}{
 		"id": s.node,
 		"stats": &nodeStats{
-			Active:       true,
-			Mining:       mining,
-			Hashrate:     hashrate,
-			Peers:        s.server.PeerCount(),
-			GasPrice:     gasprice,
-			Syncing:      syncing,
-			Uptime:       100,
-			Chain:        common.NodeLocation.Name(),
-			ChainID:      s.chainID.Uint64(),
-			LatestHeight: header.Number().Uint64(),
-			LatestHash:   header.Hash().String(),
+			Name:             s.node,
+			Active:           true,
+			Mining:           mining,
+			Hashrate:         hashrate,
+			Peers:            s.server.PeerCount(),
+			GasPrice:         gasprice,
+			Syncing:          syncing,
+			Uptime:           100,
+			Chain:            common.NodeLocation.Name(),
+			ChainID:          s.chainID.Uint64(),
+			LatestHeight:     header.Number().Uint64(),
+			LatestHash:       header.Hash().String(),
+			CPUPercentUsage:  float32(cpuPercent[0]),
```

Comment:
Why not give the full array of CPU percents? If we're collecting the data, we might as well get the different filter windows

Reply:
it's actually an array of size 1 because cpu.Percent was called with percpu set to false